File: applicationInJava/src/main/java/com/kestraa/hexagonal/repository/UserRepositoryImpl.java
package com.kestraa.hexagonal.repository;

import com.kestraa.hexagonal.model.User;
import com.kestraa.hexagonal.port.UserRepository;
import org.springframework.stereotype.Repository;

import java.util.ArrayList;
import java.util.List;

@Repository
public class UserRepositoryImpl implements UserRepository {

    private List<User> users = new ArrayList<>();

    @Override
    public List<User> getUsers() {
        return users;
    }

    @Override
    public void createUser(User user) {
        users.add(user);
    }
}

File: applicationJavalin/src/main/kotlin/com/kestraa/hexagonal/repository/UserRepositoryImpl.kt
package com.kestraa.hexagonal.repository

import com.kestraa.hexagonal.model.User
import com.kestraa.hexagonal.port.UserRepository

class UserRepositoryImpl : UserRepository {

    private val listUsers = mutableListOf<User>()
    override fun getUsers(): List<User> {
        return listUsers
    }

    override fun createUser(user: User) {
        listUsers.add(user)
    }
}